gnucash/gnome/gnc-plugin-page-account-tree.c

```diff
@@ -1499,21 +1500,51 @@ gnc_plugin_page_account_tree_cmd_delete_account (GtkAction *action, GncPluginPag
          * Note that one effect of the modal dialog is preventing
          * the account selectors from being repopulated.
          */
-        response = gtk_dialog_run(GTK_DIALOG(dialog));
-        if (GTK_RESPONSE_ACCEPT != response)
+        account_currency = gnc_account_get_currency_or_parent (account);
+        // Ideally we should check all subaccount currencies, and verify they're all identical. If not, it's not advisable
+        // to move subaccount currencies to a new account.
+        while (1)
         {
-            /* Account deletion is cancelled, so clean up and return. */
-            gtk_widget_destroy(dialog);
-            g_list_free(filter);
-            g_free(acct_name);
-            return;
+            gboolean ta_match = TRUE;
+            response = gtk_dialog_run(GTK_DIALOG(dialog));
+            if (GTK_RESPONSE_ACCEPT != response)
```

Comment:
Don't use constant-first comparisons. The claim that it prevents assignment mistakes is BS, compilers know about that and error out.